xmodule/modulestore/__init__.py

```diff
@@ -313,13 +324,54 @@ def send_pre_publish_signal(self, bulk_ops_record, course_id):
         if signal_handler and bulk_ops_record.has_publish_item:
             signal_handler.send("pre_publish", course_key=course_id)
 
+    def create_catalog_data_for_signal(self, course_id):
+        """
+        Creates a CourseCatalogData object from a course id by fetching from the module store
+
+        Arguments:
+            course_id (CourseKey): id of the course to process
+
+        Returns:
+            CourseCatalogData
+        """
+        # Most of this is copied from CourseOverview. We cannot import it
```

Comment:
Note this. Most of this code was copied because
A. we're not supposed to use CourseOverview and
B. circular import

Reply:
Mostly stolen from https://github.com/openedx/edx-platform/blob/db32ff2cdf678fa8edd12c9da76a76eef0478614/openedx/core/djangoapps/content/course_overviews/models.py#L148